N/NetCDF/build_tarballs.jl

```diff
@@ -98,13 +84,14 @@ products = [
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    Dependency(PackageSpec(name="HDF5_jll"), compat="~1.12.2"),
-    Dependency("Zlib_jll"),
-    Dependency("XML2_jll"),
+    Dependency("Bzip2_jll"),
+    Dependency("HDF5_jll"; compat = "1.14"),
```

Comment:
We might want 
```suggestion
    Dependency("HDF5_jll"; compat = "~1.14"),
```

See
```
julia> Pkg.Types.semver_spec("1.14") |> print
1.14.0-1
julia> Pkg.Types.semver_spec("~1.14.0") |> print
1.14
```